09-features.md

```diff
@@ -25,6 +25,7 @@ The Context column decodes as follows:
 * `C-`: presented in the `channel_announcement` message, but always odd (optional).
 * `C+`: presented in the `channel_announcement` message, but always even (required).
 * `9`: presented in [BOLT 11](11-payment-encoding.md) invoices.
+* `2`: presented in [BOLT 12](12-offer-encoding.md) messages.
```

Comment:
I'm not sure what phantom invoices are, but yes, MPP is assumed?

Reply:
Ah, LDK will need MPP support to be optional in `invoice` OMs at least.. I'm not sure the best way of doing this, maybe it would be easiest to put the MPP feature bit in offer features (though it has a dependency on the `payment_secret` feature currently, which is kinda awkward)?

Reply:
Wait, how do you *not* support MPP?  In 2022, that seems like more work than supporting it everywhere (we don't *send* MPP for keysend, but of course we have no problem receiving it...).

Reply:
Heh, we support MPP for regular invoices. But for multi-node receive invoices where the payment can be received by any backing node, we don't support MPP (because the nodes would have to coordinate with each other before releasing the preimage, which is not supported).

Reply:
Sorry, that's not valid :(

You have an implementation (or maybe even a design?) which doesn't work for modern payments.  That you can currently make it work at all is a historical accident.  But really, not something we want to be designing around!

Reply:
It seems that what you want is to force the sender to send everything to the same blinded path. Disabling MPP completely is a crude way of doing it, the sender could still split the payment as long as all the part have the same destination.
If you want to enable a new design, you should add a new feature for it, not disable an existing feature that's considered part of the base package.

Reply:
> It seems that what you want is to force the sender to send everything to the same blinded path.

FYI, see the following for full context: https://lightningdevkit.org/blog/introducing-phantom-node-payments/

The key point is being able to retry on a different path (i.e., failover as @TheBlueMatt mentioned) without coordination between nodes.